src/dfx/src/config/cache.rs

```diff
@@ -65,66 +58,11 @@ impl Cache for DiskBasedCache {
     }
 
     fn get_binary_command(&self, binary_name: &str) -> Result<std::process::Command, CacheError> {
+        Self::install(&self.version_str())?;
```

Comment:
Was this not happening before?

Reply:
it was happening here: https://github.com/dfinity/sdk/pull/3036/files#diff-0c4331b4ea6b0fad92e885ffcb7a58541bef24817e1d495cdc56b8266f1319b2L219
but `fn get_binary_path_from_version` got moved to `dfx-core` so it can no longer access `fn install()`, so I had to reshuffle

Reply:
Okay.  Do you think `get_binary_command_path()` in this file (5 lines above) also needs to call Self::install?

Reply:
🙈 thanks!!